custom-routes-vis/src/main/java/com/google/sps/servlets/GetDirectionsServlet.java

```
@@ -0,0 +1,120 @@
+      throws IOException, MalformedURLException {
+    String originDestination = "&origin=" + origin + "&destination=" + destination;
+    URL directionsUrl = new URL("https://maps.googleapis.com/maps/api/directions/json?"
+          + "mode=driving"
+          + "&alternatives=true" + originDestination + "&key=" + key);
+    URLConnection connection = directionsUrl.openConnection();
+    connection.setRequestProperty("Accept-Charset", "UTF-8");
+    return connection;
+  }
+
+  /** 
+   * Sends request to the Routes Preferred API and returns the connection.
+   * @param origin Origin of the requested routes in "lat,lng" format.
+   * @param destination Destination of the requested routes in "lat,lng" format.
+   * @param alpha Whether to send the request to the alpha version of the API.
+   * @param key The API key.
+   */
+  public URLConnection sendPostRequestToRoutesApi(String origin, String destination, boolean alpha,
+      String apiKey) throws IOException, MalformedURLException {
+    URL routesUrl;
+    if (alpha) {
+      routesUrl = new URL("https://routespreferred.googleapis.com/v1alpha:computeCustomRoutes");
+    } else {
+      routesUrl = new URL("https://routespreferred.googleapis.com/v1:computeRoutes");
+    }
+
+    HttpURLConnection connection = (HttpURLConnection) routesUrl.openConnection();
+    connection.setRequestMethod("POST");
+    connection.setRequestProperty("Content-Type", "application/json; utf-8");
+    connection.setRequestProperty("Accept", "application/json");
+    connection.setDoOutput(true);
+
+    String requestParamsJson = "{"
+        + "\"origin\": {\"location: \"" + origin + "\"}," 
+        + "\"destination\": {\"location: \"" + destination + "\"},"
+        + "\"travelMode\": \"DIRVE\", \"computeAlternativeRoutes\": true}";
+    try(OutputStream os = connection.getOutputStream()) {
+      byte[] input = requestParamsJson.getBytes("utf-8");
+      os.write(input, 0, input.length);			
+    }
```

Anything for the `catch` block?

---

I guess we can just throw an error if writing the request body to outputstream fails